### ISSUE TITLE
ClientsideModel render group always set to RENDERGROUP_OPAQUE

### ISSUE THREAD

**Issue by LeQuackers** [Facepunch/garrysmod-issues#4497]
As the title says, ClientsideModel always produces an entity with the render group RENDERGROUP_OPAQUE despite changing its arguments.
```
local csent1 = ClientsideModel("models/props_junk/metal_paintcan001a.mdl")
print(csent1:GetRenderGroup(), RENDERGROUP_OTHER) -- the default should be RENDERGROUP_OTHER according to the wiki

local csent2 = ClientsideModel("models/props_junk/metal_paintcan001a.mdl", RENDERGROUP_OTHER)
print(csent2:GetRenderGroup(), RENDERGROUP_OTHER)

local csent3 = ClientsideModel("models/props_junk/metal_paintcan001a.mdl", RENDERGROUP_VIEWMODEL)
print(csent3:GetRenderGroup(), RENDERGROUP_VIEWMODEL)

csent1:Remove()
csent2:Remove()
csent3:Remove()
```

**Comment by thegrb93** [comment]
I've also been meaning to report that it seems ClientsideModel is also getting physics created though I'm not entirely sure yet.

**Comment by robotboy655** [comment]
@thegrb93  ClientsideModel don't get physics, but ents.CreateClientProp will like expected.

**Comment by robotboy655** [comment]
I don't think it was ever meant to change the actual render group of the entity, you simply cannot do that directly in-engine by default. 

The render group used there is for the "clientside leaf system", whatever that means.

**Comment by robotboy655** [comment]
That being said, it may still have an effect, but GetRenderGroup will return whatever ` C_BaseEntity::GetRenderGroup` calculates internally.

**Comment by GitSparTV** [comment]
Shouldn't you set the rendergroup in the second argument? I was always using that

**Comment by Kefta** [comment]
Render groups use a callback function, not a class variable, hence why there's no SetRenderGroup.

**Comment by LeQuackers** [comment]
So I did some research and I think this is actually an oversight.

> The render group used there is for the "clientside leaf system", whatever that means.

Given that I'm assuming ```ClientsideModel``` is a wrapper for ```C_BaseEntity::InitializeAsClientEntity```.

The leaf system appears to control when/how rendering occurs for all ```IClientRenderable```, so setting the rendergroup for ```ClientsideModel``` should still work. I think the issue is ```C_BaseEntity::ComputeFxBlend``` being called inside ```CClientLeafSystem::ComputeTranslucentRenderLeaf```, which resets the rendergroup to whatever ```C_BaseEntity::GetRenderGroup``` returns (probably not the rendergroup you want).



```C_BaseEntity::ComputeFxBlend``` resetting the rendergroup:
https://github.com/ValveSoftware/source-sdk-2013/blob/master/mp/src/game/client/c_baseentity.cpp#L3512

```C_BaseEntity::ComputeFxBlend``` being called in ```CClientLeafSystem::ComputeTranslucentRenderLeaf```:
https://github.com/ValveSoftware/source-sdk-2013/blob/master/mp/src/game/client/clientleafsystem.cpp#L1408

**Comment by LeQuackers** [comment]
The best way to fix this would probably be Facepunch/garrysmod-requests/issues/1022.
